src/Microsoft.SqlTools.ServiceLayer/DisasterRecovery/BackupOperation/IBackupOperation.cs

```diff
@@ -38,7 +38,12 @@ public interface IBackupOperation
         /// Execute backup
         /// </summary>
         void PerformBackup();
```

Comment:
Minor: why not have a generic task API instead?
- `Execute(IExecuteParams config)`
- `Cancel`
- `OnCanceled` event
- `OnScriptCreated` event
- `OnComplete` event

`IExecuteParams`: options to script, execute, or both?